docs/docs/reference/kind-polymorphism.md

```diff
@@ -0,0 +1,43 @@
+---
+layout: doc-page
+title: "Kind Polymorphism"
+---
+
+Normally type parameters in Scala are partitioned into _kinds_. First-level types are types of values. Higher-kinded types are type constructurs
+such as `List` or `Map`. The kind of a type is indicated by the top type of which it is a subtype. Normal types are subtypes of `Any`,
+covariant single argument type constructors such as `List` are subtypes of `[+X] => Any`, and the `Map` type constructor is
+a subtype of `[X, +Y] => Any`.
+
+A type can be used only as prescribed by its kind. Subtypes of `Any` cannot be applied to type arguments whereas subtypes of `[X] => Any`
+_must_ be applied to a type argument, unless they are passed to type parameters of the same kind.
+
+Sometimes we would like to have type parameters that can have more than one kind, for instance to define an implicit
+value that works for parameters of any kind. This is now possible through a form of (_subtype_) kind polymorphism.
+Kind polymorphism relies on the special type `scala.AnyKind` that can be used as an upper bound of a type.
+
+```scala
+def f[T <: AnyKind] = ...
```

Comment:
So this seems to be true top parametric type that you proposed some time ago in Scala Contributors? :smile:

Reply:
Yes, it's even topper than top since it's also above type constructors 😄